### ISSUE TITLE
babeld replaces routes non-atomically

### ISSUE THREAD

**Issue by DanielG** [jech/babeld#110]
Hi Juliusz,

I've just tracked down a particularly bothersome behavior in my babeld deployment to the fact that babeld doesn't use NLM_F_REPLACE to replace changed routes atomically.

The use-case involves ingesting babel routes into bird to make use of it's support for router advertisements with (default) router lifetime controlled by IGP route presence.

What seems to happen in my system is that babeld removes the default route and quickly re-adds the modified route. Bird sees this and sends a RA lifetime=0 announcement, but it doesn't send the lifetime>0 update quite as quickly it takes a couple of seconds (not sure why yet). This obviously breaks internet connectivity for all clients on the LAN intermittently.

--Daniel

**Comment by jech** [comment]
> babeld doesn't use NLM_F_REPLACE to replace changed routes atomically.

Yep.  I did implement it many years ago, but at the time kernel-side support for it was buggy.

I think it might be a good idea to try again, perhaps the kernels have improved since then.
